x/programs/rust/examples/single_nft/src/lib.rs

```diff
@@ -0,0 +1,342 @@
+//! A basic NFT contract.
+//! The program serves as a non-fungible token with the ability to mint and burn.
+//! Only supports whole units with no decimal places.
+//!
+//! The NFT must support the common NFT metadata format.
+//! This includes the name, symbol, and URI of the NFT.
+use wasmlanche_sdk::{
+    memory::{Memory, Pointer},
+    program::Program,
+    public, state_keys,
+    types::Address,
+};
+
+/// The total supply of the NFT.
+/// # Safety
+/// In this program only one NFT of each type is intended to be minted.
+const TOTAL_SUPPLY: i64 = 1;
+
+/// The program storage keys.
+#[state_keys]
+enum StateKey {
+    /// The total supply of the token. Key prefix 0x0.
+    TotalSupply,
```

Comment:
This should be called `MaxSupply`? It does not seem to reflect the current supply (which I think is implied by `TotalSupply`)

Reply:
`MaxSupply` sounds better.